proctor-common/src/main/java/com/indeed/proctor/common/model/Payload.java

```diff
@@ -302,7 +303,85 @@ public boolean equals(final Object o) {
                 Arrays.equals(longArray, payload.longArray) &&
                 Objects.equals(stringValue, payload.stringValue) &&
                 Arrays.equals(stringArray, payload.stringArray) &&
-                Objects.equals(map, payload.map);
+                this.comparePayloadMap(payload.map);
```

Comment:
I remember you mentioned some issue motivating to introduce this compare method. Would you remind me what is the motivation?

Reply:
The issue has to do with serialize and deserializing payloads  

```
        final TestBucket bucket = TestBucket.builder()
                .name("grp1")
                .value(1)
                .description("group 1")
                .payload(new Payload(ImmutableMap.of(
                        "k", 10,
                        "v", 100.0,
                        "s", new Double[]{10.0, 20.0, 30.0}
                )))
                .build();

        final TestBucket converted = deserialize(
                serialize(bucket)
        );

        assertThat(bucket.fullEquals(converted))
                .as("%s should be fully equal to %s", bucket, converted)
                .isTrue();
```

This test will fail due to the Class type of the map. The Map<String, Object> causes issues when the value is an array. When the bucket is deserialized if the map's value is an array then it gets deserialized as a Collection List rather than an array. So in the both example it would be deserialized as "s" : Collection<Double> rathen than "s" : Double[].

Reply:
Removed.